### PR TITLE
#163426531 Like, Dislike articles and Delete User Profile 

### PR DESCRIPTION
**Pull request by primuse**
#### What does this PR do?
Have API endpoints to
- Like and Dislike an article
- Delete a User's profile (account)

#### Description of Task to be completed?
Have the following endpoints working:
- Like and Dislike an article: `POST /api/v1/articles/:slug/reaction`
- Delete a User's profile: `DELETE api/v1/profiles/:username/delete`

#### How should this be manually tested?
Clone this repo, cd into it and run npm install.
 **With test suite:**
     - Run `npm test`

 **Using postman:**
     - Run `npm run dev`,
     - With postman, send requests to `localhost:3000/<api url>` with the relevant HTTP methods and request payloads.

#### What are the relevant pivotal tracker stories?
[#163426531](https://www.pivotaltracker.com/story/show/163426531)
[#164073395](https://www.pivotaltracker.com/story/show/164073395)

### CODE REVIEW

**Comment by EfeAgare** [comment]
In the **What does this PR do** title, creates API endpoints does not really signify that the PR already have the functionality that the PR is raised for. It should be something like **Have API endpoints to**

**Review by omenkish — DISMISSED** [review]

**Review by divinediscipline — DISMISSED** [review]
Looks good to me

**Review by omenkish — APPROVED** [review]

**Review by divinediscipline — DISMISSED** [review]
Looks good to me

**Review by bobbylemm — DISMISSED** [review]
LGTM

File: server/controllers/userController.js [hunk 1]
```diff
@@ -282,16 +286,25 @@ class UserController extends BaseHelper {
...
           {
-            where: { id },
+            where: { username },
             returning: true
           }
         );
         const updatedRows = updatedUser[0];
         const updatedUserValues = updatedUser[1][0].dataValues;
+        delete updatedUserValues.password;
+        delete updatedUserValues.role;
```
Comment (p-funky) [inline server/controllers/userController.js]: Why the need for this deletion?

File: server/middlewares/validators/signUpValidator.js [hunk 2]
```diff
@@ -5,9 +5,6 @@ import errorMessage from '../../helpers/errorHelpers';
 export default (req, res, next) => {
   const { username, email, password } = req.body;
 
-  // if (!password) {
-  //   return errorMessage(res, 400, 'Password should be provided');
-  // }
```
Comment (p-funky) [inline server/middlewares/validators/signUpValidator.js:10]: Any ideas why this was commented out? Was it never hit?

File: server/middlewares/verifyUser.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import errorMessage from '../helpers/errorHelpers';
+
+
+const validateUser = (req, res, next) => {
+  const user = req.params.username;
+  const loggedUser = req.user.username;
+  if (user !== loggedUser) {
+    return errorMessage(res, 401, 'You are not authorized to access this route');
```
Comment (p-funky) [inline server/middlewares/verifyUser.js]: Should this be `401` or `403`? Is it a case of `authorization` or `authentication`?

**Review by omenkish — DISMISSED** [review]

File: server/middlewares/verifyUser.js (hunk 3 above)
Comment (p-funky) [reply server/middlewares/verifyUser.js]: And you can keep the message simple please => `User not authorized`.

File: server/tests/app/profileDelete.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      expect(res.status).to.equal(200);
+      expect(res.body.message).to.equal('Profile successfully deleted');
+    }));
+  it('should delete articles created by the deleted user', () => request(app)
+    .get('/api/v1/articles/how-i-got-into-andela')
+    .set('x-auth-token', userToken)
+    .then((res) => {
+      expect(res.status).to.equal(404);
+      expect(res.body.message).to.equal('Article not found');
+    }));
```
Comment (p-funky) [inline server/tests/app/profileDelete.test.js:62]: I like that you've checked if it deleted articles. Would be great to also check if comments and reactions (likes/dislikes) were deleted as well.

File: server/controllers/userController.js (hunk 1 above)
Comment (primuse) [reply server/controllers/userController.js]: The logic behind this is that a user is not meant to see sensitive information like password and role.

File: server/middlewares/validators/signUpValidator.js (hunk 2 above)
Comment (primuse) [reply server/middlewares/validators/signUpValidator.js:10]: I think this is as a result of the merge. I removed it in my branch, have no idea why it's still here though

File: server/middlewares/verifyUser.js (hunk 3 above)
Comment (primuse) [reply server/middlewares/verifyUser.js]: Noted @p-funky. It is meant to be a 403 (forbidden) error.

File: server/tests/app/profileDelete.test.js (hunk 4 above)
Comment (primuse) [reply server/tests/app/profileDelete.test.js:62]: Noted @p-funky

File: server/controllers/userController.js (hunk 1 above)
Comment (p-funky) [reply server/controllers/userController.js]: But you are not returning `updatedUserValues` anywhere fam. 🤷🏾‍♂️

File: server/middlewares/validators/signUpValidator.js (hunk 2 above)
Comment (p-funky) [reply server/middlewares/validators/signUpValidator.js:10]: Yes, you removed it but I am asking why it was commented out by whomever did so in the first place.

File: server/controllers/userController.js (hunk 1 above)
Comment (primuse) [reply server/controllers/userController.js]: Oh okay, I understand now. That was a previous implementation when I was returning `updatedUserValues`. I would take it out now

**Review by biolabalo — APPROVED** [review]

**Review by p-funky — APPROVED** [review]

**Review by EfeAgare — APPROVED** [review]

**Pull request merged**